src/app/rundown/components/segment/segment.component.ts

```diff
@@ -1,25 +1,103 @@
-import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core'
+import {
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  OnDestroy,
+  Output, SimpleChange,
+  SimpleChanges
+} from '@angular/core'
 import { Segment } from '../../../core/models/segment'
+import { Part } from '../../../core/models/part'
+import {PieceLayerService} from "../../../shared/services/piece-layer.service";
+import { PieceLayer } from '../../../shared/enums/piece-layer'
 
 @Component({
   selector: 'sofie-segment',
   templateUrl: './segment.component.html',
   styleUrls: ['./segment.component.scss']
 })
-export class SegmentComponent {
-
+export class SegmentComponent implements OnChanges, OnDestroy {
+  // TODO: Remove when we have change detection for reference values.
   @Input()
-  public isRundownActive: boolean
+  public isOnAir: boolean
 
   @Input()
   public segment: Segment
 
   @Output()
-  setNextEvent: EventEmitter<{segmentId: string, partId: string}> = new EventEmitter()
+  public setNextEvent: EventEmitter<{segmentId: string, partId: string}> = new EventEmitter()
+
+  public timeReference: number = 0
+  public pieceLayers: PieceLayer[] = []
+
+  private animationFrameId?: number
 
-  constructor() { }
+  constructor(private readonly pieceLayerService: PieceLayerService) {}
 
   public emitSetNextEvent(partId: string): void {
     this.setNextEvent.emit({ segmentId: this.segment.id, partId})
   }
+
+  private getUsedPieceLayersInOrder(): PieceLayer[] {
+    const pieceLayersInOrder: PieceLayer[] = this.pieceLayerService.getPieceLayersInOrder()
+    const usedPieceLayers: PieceLayer[] = this.pieceLayerService.getPieceLayersForParts(this.segment.parts)
+    return pieceLayersInOrder.filter(layer => usedPieceLayers.includes(layer))
+  }
+
+  public ngOnChanges(changes: SimpleChanges): void {
+    this.pieceLayers = this.getUsedPieceLayersInOrder()
+
+    const isOnAirChange: SimpleChange | undefined = changes['isOnAir']
+    if (isOnAirChange?.currentValue === false) {
```

Comment:
This doesn't work `!isOnAirChange?.currentValue`?

Reply:
No. If there are no changes to the `isOnAir` property, then the animation should keep on playing.